gss-serv-krb5.c

```diff
@@ -76,6 +76,52 @@ ssh_gssapi_krb5_init(void)
 	return 1;
 }
 
+/* Check if any of the indicators in the Kerberos ticket match
+ * one of indicators in the list of allowed/denied rules.
+ * In case of the match, apply the decision from the rule.
+ * If there is no match:
+ *  - if there were both denial and allow rules, deny
+ *  - if there were only denial rules, allow
+ *  - if there were only allow rules, deny
+ */
+
+static int
+ssh_gssapi_check_indicators(ssh_gssapi_client *client, int *matched)
+{
+	char *chptr;
+	int denial, allows = 0, denials = 0;
+	u_int i, j;
+
+	/* Check indicators */
+	for (i = 0; i < options.num_gss_indicators; i++) {
+		for (j = 0; client->indicators[j] != NULL; j++) {
+			chptr = strchr(options.gss_indicators[i], '=');
+			chptr[0] = '\0';
+			if (strcmp(client->indicators[j],
+			     options.gss_indicators[i]) == 0) {
+				*matched = j;
+			}
+			chptr[0] = '=';
+			denial = (chptr[1] == 'd' || chptr[1] == 'D') ? 1 : 0;
+			denials |= denial;
+			allows |= !denial;
+			if (*matched) {
+				return (1 - denial);
+			}
+		}
```

Comment:
Use `match_pattern_list` ?

Reply:
Rewrote to use `match_pattern_list`. This forced a change in the way how a rule is presented, though not too different:
```
GSSAPIIndicators [!]name1,[!]name2,...
```
Where a negation would mean a denial to the presence of the indicator.
A result is that a rule like
```
GSSAPIIndicators !hardened,pkinit,idp,passkey,otp
```
would only allow access with passwordless pre-authentication method in FreeIPA (either using smartcards, external IdP authentication, FIDO2 token or TOTP/HOTP token).

This simplifies a bit both configuration part and matching rule.

Reply:
I added this as a fixup on top of the original commit to show the difference. Let me know if this is better one. I also would like to add it to `Match` handling so that one could have per-group or per-user indicator rules. Let me know if this would be useful.